lib/mockito.dart

```diff
@@ -48,6 +53,119 @@ class Mock {
   String toString() => _givenName != null ? _givenName : runtimeType.toString();
 }
 
+// Return a new [Invocation], reconstituted from [invocation], [_typedArgs],
+// and [_typedNamedArgs].
+Invocation _reconstituteInvocation(Invocation invocation) {
+  var newInvocation = new FakeInvocation(invocation);
+  _typedArgs.clear();
```

Comment:
It looks like _typedArgs and _typedNamedArgs are being cleared without being used.
I realize now that FakeInvocation has access to them but it might be clearer to pass them into FakeInvocation to make it obvious how the invocation is reconstituted.